src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,380 @@
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /**
+     * @notice Grants `usr` operator access to this contract.
+     * @param usr The user address.
+     */
+    function hope(address usr) external auth {
+        can[usr] = 1;
+        emit Hope(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` operator access from this contract.
+     * @param usr The user address.
+     */
+    function nope(address usr) external auth {
+        can[usr] = 0;
+        emit Nope(usr);
+    }
+
+    /**
+     * @notice Whitelist `who` address for `pick`.
+     * @param who The user address.
+     */
+    function kiss(address who) external auth {
+        bud[who] = 1;
+        emit Kiss(who);
+    }
+
+    /**
+     * @notice Remove `who` address from `pick` whitelist.
+     * @param who The user address.
+     */
+    function diss(address who) external auth {
+        if (to == who) to = address(0);
+        bud[who] = 0;
+        emit Diss(who);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            require(data != address(0), "RwaOutputConduit3/invalid-quit-to-address");
+            quitTo = data;
+        } else {
+            revert("RwaOutputConduit3/unrecognised-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /**
+     * @notice Sets `who` address as the recipient.
+     * @param who Recipient address.
+     * @dev `who` address should have been whitelisted using `kiss`.
+     */
+    function pick(address who) external isMate {
+        require(bud[who] == 1 || who == address(0), "RwaOutputConduit3/not-bud");
+        to = who;
+        emit Pick(who);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Method to swap DAI contract balance to GEM through PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     */
+    function push() external isMate {
```

Comment:
consider adding a slippage check here and in the other push function for the output, to protect against unexpected fee changes. I.e add a parameter of minGemAmount. It's pretty common for swapping logic.

Reply:
Since fee changes require a governance vote and that this is a function that will not be called frequently, the chances of running into the slippage issue are pretty slim.

These conduits are meant to be operated by individuals who are **not** necessarily tech-savvy. Having a function without any parameters minimize the risk of users having the call fail because they typed one zero less than they are supposed to.

Reply:
I mean, for that to be an issue, the execution of a spell changing the PSM fees would have to "frontrun" a call to `push()`.

Changing fees would have to be signaled way before and follow Maker's governance process, so all stakeholders would be aware of it. 

Even if this unlikely condition happens in the output conduit and the lender gets less than they expect, we can always allow them to draw more Dai by increasing the Debt ceiling to cope with the difference.

If it's the same case in the input conduit, it simply means that there will still be some outstanding debt in the vault and that difference will still be incorporated to Maker's balance sheet, so by all means the debt can be considered paid-off.

Reply:
ok